src/Giraffe/WebSockets.fs

```diff
@@ -0,0 +1,182 @@
+module Giraffe.WebSocket
+
+open System
+open System.Net.WebSockets
+open System.Threading
+open System.Threading.Tasks
+open Giraffe.Tasks
+open Giraffe
+
+/// WebSocket subprotocol type. For negotiation of subprotocols.
+type WebSocketSubprotocol = {
+    /// The subprotocol name.
+    Name : string
+}
+
+/// Default WebSocket communication options.
+let DefaultWebSocketOptions = 
+    let webSocketOptions = Microsoft.AspNetCore.Builder.WebSocketOptions()
+    webSocketOptions.KeepAliveInterval <- TimeSpan.FromSeconds 120.
+    webSocketOptions.ReceiveBufferSize <- 4 * 1024
+    webSocketOptions
+
+/// Internal reference to a WebSocket. Includes WebSocketID and Subprotocol.
+type WebSocketReference = {
+    /// A reference to the WebSocket.
+    WebSocket : WebSocket
+    /// The selected subprotocol.
+    Subprotocol : WebSocketSubprotocol option
+    /// The internal ID of the WebSocket.
+    ID : string
```

Comment:
What is the benefit of using string when we are using strings of Guids, I imagine it is to allow easier transport of ID across wire?

Reply:
I guess it depends on if we want to let people choose their ids.  I could see people using a username from a jwt.

Reply:
yes we need to let people choose IDs